core/domain/exp_jobs_one_off.py

```diff
@@ -831,3 +833,115 @@ def reduce(key, values):
+            yield('Error %s when validating current exploration' % str(e),
+                  item.id)
+            return
+
+
+        states = exploration.states
+        new_states = {}
+        for state_name, state in states.iteritems():
+            state_dict = state.to_dict()
+            state_content_id_list = []
+
+            # Add state card's content id into the state_content_id_list.
+            state_content_id_list.append(state_dict['content']['content_id'])
+
+            # Add answer_groups content id into the state_content_id_list.
+            for answer_group in state_dict['interaction']['answer_groups']:
+                answer_feedback = answer_group['outcome']['feedback']
+                state_content_id_list.append(answer_feedback['content_id'])
+
+            # If present, add default_outcome content id into
+            # state_content_id_list.
+            default_outcome = state_dict['interaction']['default_outcome']
+            if default_outcome is not None:
+                state_content_id_list.append(
+                    default_outcome['feedback']['content_id'])
+
+            # Add hints content id into state_content_id_list.
+            for hint in state_dict['interaction']['hints']:
+                state_content_id_list.append(hint['hint_content']['content_id'])
+
+            # If present, add solution content id into state_content_id_list.
+            solution = state_dict['interaction']['solution']
+            if solution:
+                state_content_id_list.append(
+                    solution['explanation']['content_id'])
+
+            # Filter content_ids_to_audio_translations with unwanted
+            # content id.
+            citat = state_dict['content_ids_to_audio_translations']
+            extra_content_ids_in_citat = (
+                set(citat.keys()) - set(state_content_id_list))
+            for content_id in extra_content_ids_in_citat:
+                state_dict['content_ids_to_audio_translations'].pop(content_id)
+                yield('Deleted extra content_id from '
+                      'content_ids_to_audio_translations of %s state in %s'
+                      % (state_name, item.id), content_id)
+
+            # Create written_translations using the state_content_id_list.
+            state_dict['written_translations'] = {}
+            translations_mapping = {}
+            for content_id in state_content_id_list:
+                translations_mapping[content_id] = {}
+            state_dict['written_translations']['translations_mapping'] = (
+                translations_mapping)
+
+            if set(state_dict['content_ids_to_audio_translations'].keys()) != (
+                    set(state_content_id_list)):
+                yield ('Error when validating exploration in dict.', item.id)
+            if set(translations_mapping.keys()) != set(state_content_id_list):
+                yield ('Error when validating exploration in dict.', item.id)
```

Comment:
Try to use different error messages for each yield statement. That way, you know where the error comes from. I don't think job output provides line numbers or tracebacks.